### DISCUSSION TITLE
How to add geospatial indexes using schema.prisma?

### DISCUSSION THREAD

**Discussion by vinaykharayat** [prisma/prisma#20142] [Q&A]
I want to add a geospatial index in mongoDB using schema.prisma 
I used 
`@@index([latlng], map: "latlng_2dsphere")`
but it is adding it as regular index.
I dont want to use `db.<collection>.createIndex({ "location": "2dsphere" })`
Is there any way to achieve it using schema.prisma?

**Comment by ludralph** [comment]
Hi @vinaykharayat 👋 

Thank you for raising this question. Currently, Prisma doesn't support adding a geospatial index using the `@@index` attribute. You could add the index using the `db.<collection>.createIndex({ "location": "2dsphere" })` within a migration file. We do have an open feature request for support geospatial queries. 

- https://github.com/prisma/prisma/issues/1798

It would be helpful if you could document your use case and leave a 👍 to help our engineers prioritize this issue.

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by vinaykharayat** [reply]
As I am using mongo db I dont have migration. Also whenever I db push my schema it removes my index in mongodb then I have to create it again.
Although as you mentioned this is already a requested feature I have commented on that specific issue. I am marking this as resolved.

**Reply by ludralph** [reply]
You're right that MongoDB doesn't have migration. When I responded, I was thinking about a traditional SQL database. Thanks for clarifying.
